canonical-web-and-design/ubuntu.com
# Add cypress test - interactive forms

## Done

- Wrote cypress test for interactive forms 
- Wrote separate tests for `/cube`, `/advantage`, `/robotics`,`/openstack/pricing-calculator` each 
- Wrote separate test for pages that have email inputs both in the modal form and in the page and added `data-testid="form-email"` on the email input fields of the modal forms to distinguish it
- Added `data-testid="interactive-form"` on the CTA buttons that open interactive form 

## QA

- Check out this feature branch
- Run the site using the command `./run serve` or `dotrun`
- View the site locally in your web browser at: http://0.0.0.0:8001/

1. Create cypress.env.json file
2. Copy and paste below (Reach out to me if you don't have keys)
3. { "UBUNTU_USERNAME": "", "UBUNTU_PASSWORD": "", "MARKETO_CLIENT_ID": "", "MARKETO_CLIENT_SECRET": "", "MARKETO_AUTHORISED_USER": "", "MARKETO_TOKEN": "" }
4. Move `forms_spec.js` file in the `/tests/cypress/forms` folder to `/tests/cypress/integration` folder for testing
5. Run the site using the command ./run serve or dotrun
5. Use `yarn run cypress open --config-file tests/cypress/cypress.json --config baseUrl=http://0.0.0.0:8001/ to run cypress test`
6. Click `forms_spec.js`
7. Check if tests pass. (You can skip static contact-us page tests by adding `.skip` to `it` ) 

## Issue / Card

Fixes [#10782](https://github.com/canonical-web-and-design/ubuntu.com/issues/10782)
